offix/src/main/java/org/aerogear/offix/Offix.kt

```diff
@@ -64,8 +78,39 @@ fun ApolloClient.enqueue(
             Log.d("Extension Callback * ", result)
 
             /*
-             Set the response received from the server in the onSucceess() of responseCallback.
+             1. Check if the response data is null or not. If it's null that means the conflict has happened.
+             2. Parse the json error response from the server.
+             3. Get the serverstate and the clientstate keys from the error response.
+             4. Set them in responseCallback's onConflictDetected() method.
+             5. The user resolves conflicts and send the mutation back to the library.
+             6. Make a recursive call to the enqueue() function and retry mutation again.
              */
+            if (response.data() == null) {
+                val json = JSONObject(detectedConflict)
```

Comment:
Can we move this to a helper function and have that function return the clientState and serverState?

You can use a Pair class to return both these things from a single method.

https://stackoverflow.com/a/2832516/5471095